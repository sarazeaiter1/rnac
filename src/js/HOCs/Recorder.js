import React from 'react';
import PropTypes from 'prop-types';

import AudioUtils from '../native-interfaces/AudioUtils';
import AudioRecorder from '../native-interfaces/AudioRecorder';

class RecorderButton extends React.Component {
  static propTypes = {
    children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
    recordingOptions: PropTypes.shape({
      SampleRate: PropTypes.number,
      Channels: PropTypes.number,
      AudioEncodingBitRate: PropTypes.number,
      AudioQuality: PropTypes.string,
      AudioEncoding: PropTypes.string
    })
  };
  static defaultProps = {
    children: null,
    onStartRecording: () => null,
    onStopRecording: () => null,
    recordingDirPath: `${AudioUtils.DocumentDirectoryPath}/`,
    recordingPath: 'test.aac',
    recordingOptions: {
      SampleRate: 22050,
      Channels: 1,
      AudioQuality: 'Low',
      AudioEncoding: 'aac',
      AudioEncodingBitRate: 32000
    }
  };
  state = {
    isRecording: false
  };
  componentWillUnmount() {
    AudioRecorder.destroy();
  }

  render() {
    const { startRecording, stopRecording } = this;
    const { isRecording } = this.state;
    const { children } = this.props;
    if (!children) return null;
    return typeof this.props.children === 'function'
      ? this.props.children({ startRecording, stopRecording, isRecording })
      : React.cloneElement(this.props.children, {
          startRecording,
          stopRecording,
          isRecording
        });
  }

  startRecording = async (fileName, startRecordingConfig = {}) => {
    if (this.state.isRecording) {
      return;
    }
    const {
      recordingPath = null,
      recordingOptions = null
    } = startRecordingConfig;
    const recordingSaveToPath =
      recordingPath === null
        ? this.props.recordingDirPath + fileName
        : recordingPath;
    const mergedOptions = {
      ...this.props.recordingOptions,
      ...recordingOptions
    };
    try {
      await AudioRecorder.startRecording({
        recordingPath: recordingSaveToPath,
        recordingOptions: mergedOptions
      });
    } catch (err) {
      // eslint-disable-next-line
      console.warn('Error : ' + err.message);
    }

    this.setState({ isRecording: true });
    this.props.onStartRecording({
      recordingPath: recordingSaveToPath,
      recordingOptions: this.props.recordingOptions
    });
  };
  stopRecording = async () => {
    if (!this.state.isRecording) {
      return;
    }
    await AudioRecorder.stopRecording();
    this.setState({ isRecording: false });
    this.props.onStopRecording({
      recordingPath: this.props.recordingDirPath + this.props.recordingPath,
      recordingOptions: this.props.recordingOptions
    });
  };
}

export default RecorderButton;
