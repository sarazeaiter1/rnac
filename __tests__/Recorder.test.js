import React from 'react';
import { Text, View, Button } from 'react-native';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Recorder from '../src/js/HOCs/Recorder';

describe('Recorder', () => {
  test('exports', () => {
    expect(Recorder).toMatchSnapshot();
  });
  test('renders null with no children', () => {
    const tree = renderer.create(<Recorder start={false} />);
    expect(JSON.stringify(tree)).toEqual('null');
  });

  test('renders and accepts children as callable function', () => {
    // eslint-disable-next-line react/prop-types
    const children = ({ startRecording, stopRecording, isRecording }) => {
      return (
        <Text>
          {JSON.stringify({
            startRecording,
            stopRecording,
            isRecording
          })}
        </Text>
      );
    };
    const tree = renderer.create(
      <Recorder start={false}>
        {children}
      </Recorder>
    );
    expect(tree).toMatchSnapshot();
  });

  test('renders and accepts children as React Class Component', () => {
    class CompChildren extends React.Component {
      render() {
        // eslint-disable-next-line react/prop-types
        const { secondsElapsed } = this.props;
        return (
          <Text>
            {JSON.stringify({
              secondsElapsed
            })}
          </Text>
        );
      }
    }

    const tree = renderer.create(
      <Recorder start={false}>
        <CompChildren />
      </Recorder>
    );
    expect(tree).toMatchSnapshot();
  });

  test('same ui tree if component or function', () => {
    // eslint-disable-next-line react/prop-types
    const children = ({ secondsElapsed }) => {
      return (
        <Text>
          {JSON.stringify({
            secondsElapsed
          })}
        </Text>
      );
    };

    class CompChildren extends React.Component {
      render() {
        // eslint-disable-next-line react/prop-types
        const { secondsElapsed } = this.props;
        return (
          <Text>
            {JSON.stringify({
              secondsElapsed
            })}
          </Text>
        );
      }
    }

    const withClass = renderer.create(
      <Recorder start={false}>
        <CompChildren />
      </Recorder>
    );

    const withFunction = renderer.create(
      <Recorder start={false}>
        {children}
      </Recorder>
    );

    expect(JSON.stringify(withClass)).toEqual(JSON.stringify(withFunction));
  });
  test('test component will unMount', () => {
    const Comp = shallow(
      <Recorder recordingPath={'test.aac'}>
        {({ isRecording, startRecording, stopRecording }) => {
          return (
            <View>
              <Button
                onPress={() => {
                  if (!isRecording) {
                    return startRecording('test.aac');
                  }
                  return stopRecording();
                }}
                style={{}}
                title={isRecording ? 'Stop Recording' : 'Start Recording'}
              />
            </View>
          );
        }}
      </Recorder>
    );
    Comp.renderer._instance._instance.componentWillUnmount();
    expect(Comp).toMatchSnapshot();
  });

  test('test on start', () => {
    const Comp = shallow(
      <Recorder recordingPath={'test.aac'}>
        {({ isRecording, startRecording, stopRecording }) => {
          return (
            <View>
              <Button
                onPress={() => {
                  if (!isRecording) {
                    return startRecording('test.aac');
                  }
                  return stopRecording();
                }}
                style={{}}
                title={isRecording ? 'Stop Recording' : 'Start Recording'}
              />
            </View>
          );
        }}
      </Recorder>
    );
    Comp.renderer._instance._instance.state.isRecording = true;
    Comp.renderer._instance._instance.startRecording();
    expect(Comp).toMatchSnapshot();
  });
  test('test start recording with path', () => {
    const Comp = shallow(
      <Recorder recordingPath={'test.aac'}>
        {({ isRecording, startRecording, stopRecording }) => {
          return (
            <View>
              <Button
                onPress={() => {
                  if (!isRecording) {
                    return startRecording('test.aac');
                  }
                  return stopRecording();
                }}
                style={{}}
                title={isRecording ? 'Stop Recording' : 'Start Recording'}
              />
            </View>
          );
        }}
      </Recorder>
    );
    Comp.renderer._instance._instance.startRecording('test.aac');
    expect(Comp).toMatchSnapshot();
  });
  test('test start recording with path and configuration', () => {
    const Comp = shallow(
      <Recorder recordingPath={'test.aac'}>
        {({ isRecording, startRecording, stopRecording }) => {
          return (
            <View>
              <Button
                onPress={() => {
                  if (!isRecording) {
                    return startRecording('test.aac');
                  }
                  return stopRecording();
                }}
                style={{}}
                title={isRecording ? 'Stop Recording' : 'Start Recording'}
              />
            </View>
          );
        }}
      </Recorder>
    );
    Comp.renderer._instance._instance.startRecording('test.aac', {
      recordingPath: 'et3/',
      recordingOptions: {
        SampleRate: 44100.0,
        Channels: 2,
        AudioQuality: 'High',
        AudioEncoding: 'ima4',
        OutputFormat: 'mpeg_4',
        MeteringEnabled: false,
        AudioEncodingBitRate: 32000
      }
    });
    expect(Comp).toMatchSnapshot();
  });
  test('test stop recording while is recording ', () => {
    const Comp = shallow(
      <Recorder recordingPath={'test.aac'}>
        {({ isRecording, startRecording, stopRecording }) => {
          return (
            <View>
              <Button
                onPress={() => {
                  if (!isRecording) {
                    return startRecording('test.aac');
                  }
                  return stopRecording();
                }}
                style={{}}
                title={isRecording ? 'Stop Recording' : 'Start Recording'}
              />
            </View>
          );
        }}
      </Recorder>
    );
    Comp.renderer._instance._instance.state.isRecording = true;
    Comp.renderer._instance._instance.stopRecording();
    expect(Comp).toMatchSnapshot();
  });
  test('test stop recording while not recording ', () => {
    const Comp = shallow(
      <Recorder recordingPath={'test.aac'}>
        {({ isRecording, startRecording, stopRecording }) => {
          return (
            <View>
              <Button
                onPress={() => {
                  if (!isRecording) {
                    return startRecording('test.aac');
                  }
                  return stopRecording();
                }}
                style={{}}
                title={isRecording ? 'Stop Recording' : 'Start Recording'}
              />
            </View>
          );
        }}
      </Recorder>
    );
    Comp.renderer._instance._instance.stopRecording();
    expect(Comp).toMatchSnapshot();
  });
});
