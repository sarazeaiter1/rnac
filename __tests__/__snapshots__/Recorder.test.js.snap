// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recorder exports 1`] = `[Function]`;

exports[`Recorder renders and accepts children as React Class Component 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  {}
</Text>
`;

exports[`Recorder renders and accepts children as callable function 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  {"isRecording":false}
</Text>
`;

exports[`Recorder test component will unMount 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View>
    <Button
        onPress={[Function]}
        style={Object {}}
        title="Start Recording"
    />
</View>,
  "nodes": Array [
    <View>
      <Button
            onPress={[Function]}
            style={Object {}}
            title="Start Recording"
      />
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <RecorderButton
        onStartRecording={[Function]}
        onStopRecording={[Function]}
        recordingDirPath="undefined/"
        recordingOptions={
                Object {
                        "AudioEncoding": "aac",
                        "AudioEncodingBitRate": 32000,
                        "AudioQuality": "Low",
                        "Channels": 1,
                        "SampleRate": 22050,
                      }
        }
        recordingPath="test.aac"
>
        [Function]
</RecorderButton>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": RecorderButton {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": [Function],
          "onStartRecording": [Function],
          "onStopRecording": [Function],
          "recordingDirPath": "undefined/",
          "recordingOptions": Object {
            "AudioEncoding": "aac",
            "AudioEncodingBitRate": 32000,
            "AudioQuality": "Low",
            "Channels": 1,
            "SampleRate": 22050,
          },
          "recordingPath": "test.aac",
        },
        "refs": Object {},
        "startRecording": [Function],
        "state": Object {
          "isRecording": false,
        },
        "stopRecording": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View>
          <Button
                    onPress={[Function]}
                    style={Object {}}
                    title="Start Recording"
          />
</View>,
        "_debugID": 2,
        "_renderedOutput": <View>
          <Button
                    onPress={[Function]}
                    style={Object {}}
                    title="Start Recording"
          />
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <RecorderButton
    onStartRecording={[Function]}
    onStopRecording={[Function]}
    recordingDirPath="undefined/"
    recordingOptions={
        Object {
            "AudioEncoding": "aac",
            "AudioEncodingBitRate": 32000,
            "AudioQuality": "Low",
            "Channels": 1,
            "SampleRate": 22050,
          }
    }
    recordingPath="test.aac"
>
    [Function]
</RecorderButton>,
}
`;

exports[`Recorder test on start 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View>
    <Button
        onPress={[Function]}
        style={Object {}}
        title="Start Recording"
    />
</View>,
  "nodes": Array [
    <View>
      <Button
            onPress={[Function]}
            style={Object {}}
            title="Start Recording"
      />
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <RecorderButton
        onStartRecording={[Function]}
        onStopRecording={[Function]}
        recordingDirPath="undefined/"
        recordingOptions={
                Object {
                        "AudioEncoding": "aac",
                        "AudioEncodingBitRate": 32000,
                        "AudioQuality": "Low",
                        "Channels": 1,
                        "SampleRate": 22050,
                      }
        }
        recordingPath="test.aac"
>
        [Function]
</RecorderButton>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": RecorderButton {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": [Function],
          "onStartRecording": [Function],
          "onStopRecording": [Function],
          "recordingDirPath": "undefined/",
          "recordingOptions": Object {
            "AudioEncoding": "aac",
            "AudioEncodingBitRate": 32000,
            "AudioQuality": "Low",
            "Channels": 1,
            "SampleRate": 22050,
          },
          "recordingPath": "test.aac",
        },
        "refs": Object {},
        "startRecording": [Function],
        "state": Object {
          "isRecording": true,
        },
        "stopRecording": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View>
          <Button
                    onPress={[Function]}
                    style={Object {}}
                    title="Start Recording"
          />
</View>,
        "_debugID": 4,
        "_renderedOutput": <View>
          <Button
                    onPress={[Function]}
                    style={Object {}}
                    title="Start Recording"
          />
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <RecorderButton
    onStartRecording={[Function]}
    onStopRecording={[Function]}
    recordingDirPath="undefined/"
    recordingOptions={
        Object {
            "AudioEncoding": "aac",
            "AudioEncodingBitRate": 32000,
            "AudioQuality": "Low",
            "Channels": 1,
            "SampleRate": 22050,
          }
    }
    recordingPath="test.aac"
>
    [Function]
</RecorderButton>,
}
`;

exports[`Recorder test start recording with path 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View>
    <Button
        onPress={[Function]}
        style={Object {}}
        title="Start Recording"
    />
</View>,
  "nodes": Array [
    <View>
      <Button
            onPress={[Function]}
            style={Object {}}
            title="Start Recording"
      />
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <RecorderButton
        onStartRecording={[Function]}
        onStopRecording={[Function]}
        recordingDirPath="undefined/"
        recordingOptions={
                Object {
                        "AudioEncoding": "aac",
                        "AudioEncodingBitRate": 32000,
                        "AudioQuality": "Low",
                        "Channels": 1,
                        "SampleRate": 22050,
                      }
        }
        recordingPath="test.aac"
>
        [Function]
</RecorderButton>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": RecorderButton {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": [Function],
          "onStartRecording": [Function],
          "onStopRecording": [Function],
          "recordingDirPath": "undefined/",
          "recordingOptions": Object {
            "AudioEncoding": "aac",
            "AudioEncodingBitRate": 32000,
            "AudioQuality": "Low",
            "Channels": 1,
            "SampleRate": 22050,
          },
          "recordingPath": "test.aac",
        },
        "refs": Object {},
        "startRecording": [Function],
        "state": Object {
          "isRecording": false,
        },
        "stopRecording": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View>
          <Button
                    onPress={[Function]}
                    style={Object {}}
                    title="Start Recording"
          />
</View>,
        "_debugID": 6,
        "_renderedOutput": <View>
          <Button
                    onPress={[Function]}
                    style={Object {}}
                    title="Start Recording"
          />
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <RecorderButton
    onStartRecording={[Function]}
    onStopRecording={[Function]}
    recordingDirPath="undefined/"
    recordingOptions={
        Object {
            "AudioEncoding": "aac",
            "AudioEncodingBitRate": 32000,
            "AudioQuality": "Low",
            "Channels": 1,
            "SampleRate": 22050,
          }
    }
    recordingPath="test.aac"
>
    [Function]
</RecorderButton>,
}
`;

exports[`Recorder test start recording with path and configuration 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View>
    <Button
        onPress={[Function]}
        style={Object {}}
        title="Start Recording"
    />
</View>,
  "nodes": Array [
    <View>
      <Button
            onPress={[Function]}
            style={Object {}}
            title="Start Recording"
      />
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <RecorderButton
        onStartRecording={[Function]}
        onStopRecording={[Function]}
        recordingDirPath="undefined/"
        recordingOptions={
                Object {
                        "AudioEncoding": "aac",
                        "AudioEncodingBitRate": 32000,
                        "AudioQuality": "Low",
                        "Channels": 1,
                        "SampleRate": 22050,
                      }
        }
        recordingPath="test.aac"
>
        [Function]
</RecorderButton>,
      "_debugID": 7,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": RecorderButton {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": [Function],
          "onStartRecording": [Function],
          "onStopRecording": [Function],
          "recordingDirPath": "undefined/",
          "recordingOptions": Object {
            "AudioEncoding": "aac",
            "AudioEncodingBitRate": 32000,
            "AudioQuality": "Low",
            "Channels": 1,
            "SampleRate": 22050,
          },
          "recordingPath": "test.aac",
        },
        "refs": Object {},
        "startRecording": [Function],
        "state": Object {
          "isRecording": false,
        },
        "stopRecording": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 4,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View>
          <Button
                    onPress={[Function]}
                    style={Object {}}
                    title="Start Recording"
          />
</View>,
        "_debugID": 8,
        "_renderedOutput": <View>
          <Button
                    onPress={[Function]}
                    style={Object {}}
                    title="Start Recording"
          />
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <RecorderButton
    onStartRecording={[Function]}
    onStopRecording={[Function]}
    recordingDirPath="undefined/"
    recordingOptions={
        Object {
            "AudioEncoding": "aac",
            "AudioEncodingBitRate": 32000,
            "AudioQuality": "Low",
            "Channels": 1,
            "SampleRate": 22050,
          }
    }
    recordingPath="test.aac"
>
    [Function]
</RecorderButton>,
}
`;

exports[`Recorder test stop recording while is recording  1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View>
    <Button
        onPress={[Function]}
        style={Object {}}
        title="Start Recording"
    />
</View>,
  "nodes": Array [
    <View>
      <Button
            onPress={[Function]}
            style={Object {}}
            title="Start Recording"
      />
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <RecorderButton
        onStartRecording={[Function]}
        onStopRecording={[Function]}
        recordingDirPath="undefined/"
        recordingOptions={
                Object {
                        "AudioEncoding": "aac",
                        "AudioEncodingBitRate": 32000,
                        "AudioQuality": "Low",
                        "Channels": 1,
                        "SampleRate": 22050,
                      }
        }
        recordingPath="test.aac"
>
        [Function]
</RecorderButton>,
      "_debugID": 9,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": RecorderButton {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": [Function],
          "onStartRecording": [Function],
          "onStopRecording": [Function],
          "recordingDirPath": "undefined/",
          "recordingOptions": Object {
            "AudioEncoding": "aac",
            "AudioEncodingBitRate": 32000,
            "AudioQuality": "Low",
            "Channels": 1,
            "SampleRate": 22050,
          },
          "recordingPath": "test.aac",
        },
        "refs": Object {},
        "startRecording": [Function],
        "state": Object {
          "isRecording": true,
        },
        "stopRecording": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 5,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View>
          <Button
                    onPress={[Function]}
                    style={Object {}}
                    title="Start Recording"
          />
</View>,
        "_debugID": 10,
        "_renderedOutput": <View>
          <Button
                    onPress={[Function]}
                    style={Object {}}
                    title="Start Recording"
          />
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <RecorderButton
    onStartRecording={[Function]}
    onStopRecording={[Function]}
    recordingDirPath="undefined/"
    recordingOptions={
        Object {
            "AudioEncoding": "aac",
            "AudioEncodingBitRate": 32000,
            "AudioQuality": "Low",
            "Channels": 1,
            "SampleRate": 22050,
          }
    }
    recordingPath="test.aac"
>
    [Function]
</RecorderButton>,
}
`;

exports[`Recorder test stop recording while not recording  1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View>
    <Button
        onPress={[Function]}
        style={Object {}}
        title="Start Recording"
    />
</View>,
  "nodes": Array [
    <View>
      <Button
            onPress={[Function]}
            style={Object {}}
            title="Start Recording"
      />
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <RecorderButton
        onStartRecording={[Function]}
        onStopRecording={[Function]}
        recordingDirPath="undefined/"
        recordingOptions={
                Object {
                        "AudioEncoding": "aac",
                        "AudioEncodingBitRate": 32000,
                        "AudioQuality": "Low",
                        "Channels": 1,
                        "SampleRate": 22050,
                      }
        }
        recordingPath="test.aac"
>
        [Function]
</RecorderButton>,
      "_debugID": 11,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": RecorderButton {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": [Function],
          "onStartRecording": [Function],
          "onStopRecording": [Function],
          "recordingDirPath": "undefined/",
          "recordingOptions": Object {
            "AudioEncoding": "aac",
            "AudioEncodingBitRate": 32000,
            "AudioQuality": "Low",
            "Channels": 1,
            "SampleRate": 22050,
          },
          "recordingPath": "test.aac",
        },
        "refs": Object {},
        "startRecording": [Function],
        "state": Object {
          "isRecording": false,
        },
        "stopRecording": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 6,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View>
          <Button
                    onPress={[Function]}
                    style={Object {}}
                    title="Start Recording"
          />
</View>,
        "_debugID": 12,
        "_renderedOutput": <View>
          <Button
                    onPress={[Function]}
                    style={Object {}}
                    title="Start Recording"
          />
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <RecorderButton
    onStartRecording={[Function]}
    onStopRecording={[Function]}
    recordingDirPath="undefined/"
    recordingOptions={
        Object {
            "AudioEncoding": "aac",
            "AudioEncodingBitRate": 32000,
            "AudioQuality": "Low",
            "Channels": 1,
            "SampleRate": 22050,
          }
    }
    recordingPath="test.aac"
>
    [Function]
</RecorderButton>,
}
`;
